go/enclave/rollupchain/rollup_chain.go

```diff
@@ -106,17 +106,13 @@ func (rc *RollupChain) ProduceNewRollup(blockHash *common.L1RootHash) (*common.E
 	if err != nil {
 		return nil, fmt.Errorf("could not retrieve block to produce rollup. Cause: %w", err)
 	}
-	chainHeads, err := rc.storage.FetchHeadsAfterL1Block(*blockHash)
+	l2Head, err := rc.storage.FetchL2Head(*blockHash)
 	if err != nil {
-		if errors.Is(err, errutil.ErrNotFound) {
```

Comment:
With the unbundling of L1 block ingestion and rollup production, this check didn't make sense.